### PR TITLE
JSONRPC validator

### PR DESCRIPTION
**Pull request by riptl**
Implements #3 :)

### CODE REVIEW

File: woocommerce-gateway-nimiq.php [hunk 1]
```diff
@@ -245,6 +246,16 @@ public function init_form_fields() {
 					'default'     => __( 'You will receive email updates after your payment has been confirmed and when we sent your order.' ),
 					'desc_tip'    => true,
 				),
+
+				// Conditional fields
+
+				'url' => array(
+					'title'       => __( 'Validation Service URL', 'wc-gateway-nimiq' ),
+					'type'        => 'textarea',
```
Comment (sisou) [inline woocommerce-gateway-nimiq.php]: 1. I would like to see the name of the setting be prefixed with the service type it is for, like `json-rpc_url`.
2. The setting should come directly after the service selector setting IMHO. 
3. It should be a `text` type, as the url is a single line setting.

File: js/settings.js [hunk 2]
```diff
@@ -6,13 +6,17 @@
 
         // Disable all conditional fields
         var conditional_fields = [
+            '#url',
```
Comment (sisou) [inline js/settings.js]: This is not the correct ID as generated by woocommerce in the settings page.

**Review by sisou — CHANGES_REQUESTED** [review]
Thank you for this! I added some improvement suggestions.

File: validation_services/json_rpc.php [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    public function value() {
+        return $this->transaction->result->value;
+    }
+
+    /**
+     * Returns the data (message) of the transaction in plain text
+     * @return {string}
+     */
+    public function message() {
+        if ( $this->transaction->data === null ) {
```
Comment (sisou) [inline validation_services/json_rpc.php:120]: Every other property is behind the `->result` property. I guess `data` should too?

**Review by sisou — CHANGES_REQUESTED** [review]

File: js/settings.js (hunk 2 above)
Comment (riptl) [reply js/settings.js]: Good catch!

**Review by sisou — APPROVED** [review]

**Pull request merged**